ZEngine/ZEngine/Core/Container/Vec.h

```diff
@@ -0,0 +1,304 @@
+#pragma once
+#include <Allocator.h>
+#include <Logging/LoggerDefinition.h>
+#include <type_traits>
+
+// using span ?
+
+using namespace ZEngine::Core::Memory;
+
+namespace ZEngine::Core::Container
+{
+
+    template <typename T>
+    class Vec
+    {
+    public:
+        using value_type      = T;
+        using size_type       = size_t;
+        using difference_type = ptrdiff_t;
+        using reference       = T&;
+        using const_reference = const T&;
+        using pointer         = T*;
+        using const_pointer   = const T*;
+        using iterator        = T*;
+        using const_iterator  = const T*;
+
+        Vec(Memory::ArenaAllocator& allocator) : m_allocator(allocator), m_size(0), m_capacity(0), m_data(nullptr) {}
+
+        Vec(Memory::ArenaAllocator& allocator, size_type initial_capacity) : m_allocator(allocator), m_size(0), m_capacity(0), m_data(nullptr)
+        {
+            reserve(initial_capacity);
+        }
```

Comment:
All ctors should be parameter less.
We want to introduce an Init(...) or Initialize(...) to accept those parameters